Pull Request for major changes to Tests and then the small change in tvl2018 that fixes the issue in signal segment to spectrum

jerhsiao opened the pull request.

jerhsiao committed:
Added tests to compare with current expected results with a set Class test parameters of 1khz, 100ms, 32000 rate, and 50 db_max, so you can't change the other function test parameters because it is being compared to a specific set of expected answers .I know this isn't the most optimal way to compare expected values or write tests, but I think it is useful for future users who are making changes and want to maintain the same exact functionality as the original loudness model.

jerhsiao committed:
Merge branch 'main' of https://github.com/jerhsiao/LoudnessModel

jerhsiao committed:
Updating tests

MalcolmSlaney commented:
Ok, that is a lot of changes.

I thought it was mostly working and you were going to add one more test to show the stuff I broke.  Perhaps just compute the values in the text summary.

It looks like you did that part differently than I did.  Did you find my error?

What do the new tests do?  (I’ll have to port them to the JAX version after we finalize this PR.)

Can you tell me more what is changing?

Thank you!

- Malcolm


> On Oct 7, 2024, at 11:43 AM, Jeremy Hsiao ***@***.***> wrote:
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/MalcolmSlaney/LoudnessModel/pull/4
> 
> Commit Summary
> 
> bd9b913 <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/bd9b9133cee96fc8cd1970d684ca7f2802d11d70> Added tests to compare with current expected results with a set Class test parameters of 1khz, 100ms, 32000 rate, and 50 db_max, so you can't change the other function test parameters because it is being compared to a specific set of expected answers .I know this isn't the most optimal way to compare expected values or write tests, but I think it is useful for future users who are making changes and want to maintain the same exact functionality as the original loudness model.
> a5a0513 <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/a5a0513f7dd770a2fcbe61b51f69d99d29e1fc32> Merge branch 'main' of https://github.com/jerhsiao/LoudnessModel
> 7b2b58e <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/7b2b58ed822da2f9c7a081c32743efde24410f34> Updating tests
> File Changes (18 files <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files>)
> D results/synthesize_1khz_100ms_50dB_calibration_level_TVL_2018_jax.txt <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-409e7b8c26f9dce0c0123b25beeecbb43b262243bc8244afd0345db29fc98658> (121)
> D results/synthesize_1khz_100ms_50dB_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-a96395e6127bf55a1b07cdad64171857c69e842db897e87fa1bdcf9ea96d0a56> (0)
> D results/test_agc_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-93510653d5cb460080e9253d3df676c353ebfe47d39591b504242b7bb6973e7e> (0)
> D results/test_agc_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-efc6cb2ead4134300a431402daa93dbf382c3e09fe7bc63e163fae87dac46da9> (0)
> D results/test_basic_example_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-e3d972e77b8e0adb0ab5186dc686f103a3a15e4325a97252046f8f3a1c44491b> (0)
> D results/test_instantaneous_specific_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0a5c5d6961341aa623eeba4c682eaec27026ee037548008e5f3bbca55a3d2ed> (0)
> D results/test_instantaneous_specific_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0c1669452ee35a9b93fb9a3883d7cecebe78661c3b1a789404430af94ea9eab> (0)
> M results/test_interpolation_linear.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-cdc9e3f7eaca0053a9550ddccac9529500f1da4c2525e9713ff07a34a35fb21d> (0)
> D results/test_interpolation_linear_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-fcefec386a4296fed54cb11426b044618457bf6e43e65c94a7d93d486b8ceccc> (0)
> M results/test_interpolation_pchip.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-3e23be5795a1a27cac20dfb48b2bc56859b7d195686966d33010458d6b2ae6de> (0)
> D results/test_interpolation_pchip_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-04645a214eb254aefd28fb82cbcfaedf75a912e634b63d7afee25c4e3be823db> (0)
> D results/test_longterm_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0f3e3f70aed18911bdb0adfde81037ef546c20e2585e6f88f870326cf2b39d2> (0)
> D results/test_longterm_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-6d87dd309e29fe6c7e7e1920c9e8e9f9c9daa13753962efb04b60d4de715bb5c> (0)
> D results/test_main_tv2018_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-60ef01a767c6dc08381c9201eb2666157ae71a0c455ed2013437b0dc1dc0edea> (0)
> M results/test_synthesize_sound_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-156d24aea1656e22610bb20ff34f8a8d9dcdd837c41e789c28365ff2a1340e65> (0)
> D results/test_synthesize_sound_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-103ce63b561707a80376da43c4ec5fab03784b58ef74105636db261ef22a52f8> (0)
> M tvl2018.py <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-c2d24759129119c8945c4911e505a80d5079aa846266ad5b20de92d35da0ade2> (20)
> M tvl2018_test.py <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-9032e90eab4f0f75c326ac02685454ad5e188d46d9fbee6a9a1c3e1808310d54> (709)
> Patch Links:
> 
> https://github.com/MalcolmSlaney/LoudnessModel/pull/4.patch
> https://github.com/MalcolmSlaney/LoudnessModel/pull/4.diff
> —
> Reply to this email directly, view it on GitHub <https://github.com/MalcolmSlaney/LoudnessModel/pull/4>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAHKLJZUL63GDEH7PEU53W3Z2LI5JAVCNFSM6AAAAABPQRUQHOVHI2DSMVQWIX3LMV43ASLTON2WKOZSGU3TCMJYHA2DGNY>.
> You are receiving this because you are subscribed to this thread.
>

jerhsiao commented:
It is working!
The main program code is mostly the same except for a few lines of code. I described what changed in my last email. I think the error had to do with using numpy ones and zeros, which were messing up the scale somehow. It might have been selecting the wrong window. I changed it to booleans that are set to a specific window and it is more closely aligned to the original MATLAB code.

As for the tests, they look like a lot of changes, but I kept interpolation tests and other small function tests the same, and just moved things around. Previously, I think the passing threshold was too high; it was just checking shapes and low level functionality. I wanted to add tests that could actually be useful for future users if they want to make more changes to the code.

Basically what I did was I found the expected loudness values for the set input of 50 max dbl, 32k rate, 1khz100ms with free field transform. I also found other expected values for each bigger function like signal segment to spectrum or the other excitation functions.
Then I added tests to check the current value against this expected. It looks like a lot but it’s pretty basic and is useful if a user makes changes and wants to compare to a version they know is correct.

Let me know if this makes sense.
Thanks!
Sincerely,
Jeremy Hsiao


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Malcolm Slaney ***@***.***>
Sent: Tuesday, October 8, 2024 6:37 AM
To: MalcolmSlaney/LoudnessModel ***@***.***>
Cc: Jeremy Hsiao ***@***.***>; Author ***@***.***>
Subject: Re: [MalcolmSlaney/LoudnessModel] Pull Request for major changes to Tests and then the small change in tvl2018 that fixes the issue in signal segment to spectrum (PR #4)


Ok, that is a lot of changes.

I thought it was mostly working and you were going to add one more test to show the stuff I broke. Perhaps just compute the values in the text summary.

It looks like you did that part differently than I did. Did you find my error?

What do the new tests do? (I’ll have to port them to the JAX version after we finalize this PR.)

Can you tell me more what is changing?

Thank you!

- Malcolm


> On Oct 7, 2024, at 11:43 AM, Jeremy Hsiao ***@***.***> wrote:
>
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/MalcolmSlaney/LoudnessModel/pull/4
>
> Commit Summary
>
> bd9b913 <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/bd9b9133cee96fc8cd1970d684ca7f2802d11d70> Added tests to compare with current expected results with a set Class test parameters of 1khz, 100ms, 32000 rate, and 50 db_max, so you can't change the other function test parameters because it is being compared to a specific set of expected answers .I know this isn't the most optimal way to compare expected values or write tests, but I think it is useful for future users who are making changes and want to maintain the same exact functionality as the original loudness model.
> a5a0513 <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/a5a0513f7dd770a2fcbe61b51f69d99d29e1fc32> Merge branch 'main' of https://github.com/jerhsiao/LoudnessModel
> 7b2b58e <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/commits/7b2b58ed822da2f9c7a081c32743efde24410f34> Updating tests
> File Changes (18 files <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files>)
> D results/synthesize_1khz_100ms_50dB_calibration_level_TVL_2018_jax.txt <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-409e7b8c26f9dce0c0123b25beeecbb43b262243bc8244afd0345db29fc98658> (121)
> D results/synthesize_1khz_100ms_50dB_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-a96395e6127bf55a1b07cdad64171857c69e842db897e87fa1bdcf9ea96d0a56> (0)
> D results/test_agc_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-93510653d5cb460080e9253d3df676c353ebfe47d39591b504242b7bb6973e7e> (0)
> D results/test_agc_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-efc6cb2ead4134300a431402daa93dbf382c3e09fe7bc63e163fae87dac46da9> (0)
> D results/test_basic_example_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-e3d972e77b8e0adb0ab5186dc686f103a3a15e4325a97252046f8f3a1c44491b> (0)
> D results/test_instantaneous_specific_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0a5c5d6961341aa623eeba4c682eaec27026ee037548008e5f3bbca55a3d2ed> (0)
> D results/test_instantaneous_specific_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0c1669452ee35a9b93fb9a3883d7cecebe78661c3b1a789404430af94ea9eab> (0)
> M results/test_interpolation_linear.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-cdc9e3f7eaca0053a9550ddccac9529500f1da4c2525e9713ff07a34a35fb21d> (0)
> D results/test_interpolation_linear_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-fcefec386a4296fed54cb11426b044618457bf6e43e65c94a7d93d486b8ceccc> (0)
> M results/test_interpolation_pchip.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-3e23be5795a1a27cac20dfb48b2bc56859b7d195686966d33010458d6b2ae6de> (0)
> D results/test_interpolation_pchip_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-04645a214eb254aefd28fb82cbcfaedf75a912e634b63d7afee25c4e3be823db> (0)
> D results/test_longterm_loudness_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-b0f3e3f70aed18911bdb0adfde81037ef546c20e2585e6f88f870326cf2b39d2> (0)
> D results/test_longterm_loudness_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-6d87dd309e29fe6c7e7e1920c9e8e9f9c9daa13753962efb04b60d4de715bb5c> (0)
> D results/test_main_tv2018_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-60ef01a767c6dc08381c9201eb2666157ae71a0c455ed2013437b0dc1dc0edea> (0)
> M results/test_synthesize_sound_plot.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-156d24aea1656e22610bb20ff34f8a8d9dcdd837c41e789c28365ff2a1340e65> (0)
> D results/test_synthesize_sound_plot_jax.png <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-103ce63b561707a80376da43c4ec5fab03784b58ef74105636db261ef22a52f8> (0)
> M tvl2018.py <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-c2d24759129119c8945c4911e505a80d5079aa846266ad5b20de92d35da0ade2> (20)
> M tvl2018_test.py <https://github.com/MalcolmSlaney/LoudnessModel/pull/4/files#diff-9032e90eab4f0f75c326ac02685454ad5e188d46d9fbee6a9a1c3e1808310d54> (709)
> Patch Links:
>
> https://github.com/MalcolmSlaney/LoudnessModel/pull/4.patch
> https://github.com/MalcolmSlaney/LoudnessModel/pull/4.diff
> —
> Reply to this email directly, view it on GitHub <https://github.com/MalcolmSlaney/LoudnessModel/pull/4>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAHKLJZUL63GDEH7PEU53W3Z2LI5JAVCNFSM6AAAAABPQRUQHOVHI2DSMVQWIX3LMV43ASLTON2WKOZSGU3TCMJYHA2DGNY>.
> You are receiving this because you are subscribed to this thread.
>


—
Reply to this email directly, view it on GitHub<https://github.com/MalcolmSlaney/LoudnessModel/pull/4#issuecomment-2398815629>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BGW4XZN5BJNRB7OJJYVCZQLZ2NOQHAVCNFSM6AAAAABPQRUQHOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOJYHAYTKNRSHE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
